Lib/http/server.py

```diff
@@ -280,9 +280,9 @@ def parse_request(self):
         self.request_version = version = self.default_request_version
         self.close_connection = True
         requestline = str(self.raw_requestline, 'iso-8859-1')
-        requestline = requestline.rstrip('\r\n')
+        requestline = requestline.rstrip(' \r\n')
         self.requestline = requestline
-        words = requestline.split()
+        words = requestline.split(' ')
```

Comment:
A different fix (leaving requestline unchanged) would be: words = requestline.strip(' ').split(' ')

But I didn't the HTTP RFC, so I don't know the exact grammar here.